arch/posix/Linux.aarch64.cmake

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# For Aarch64, multilib is not an actively pursued solution for most Linux
+# distributions. Userspace is (generally) either 32-bit or 64-bit but not
+# both.
+
+# @Intent: Call a script to get userspace wordsize for comparison with CONFIG_64BIT
+execute_process(
+  COMMAND
+  ${PYTHON_EXECUTABLE}
+  ${ZEPHYR_BASE}/scripts/user_wordsize.py
+  OUTPUT_VARIABLE
+  WORDSIZE
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+if (CONFIG_64BIT)
+  if (${WORDSIZE} STREQUAL "32")
+    message(FATAL_ERROR
+      "CONFIG_64BIT=y but this Aarch64 machine has a 32-bit userspace.\n"
+      "If using cmake use the option '-DCONFIG_BOARD_NATIVE_POSIX_64BIT=n'\n"
```

Comment:
This recommendation is not too appropriate. 2 reasons:
a) if they are compiling for native_posix, it would be better to ask them to select the target board (here `-DBOARD=native_posix_64`) instead of changing the underlying kconfig options
b) But anyhow, this is an ARCH cmake file (shared between several boards), so you do not know if they are targeting native_posix at all.
=> Maybe better change the comment to 
"If you were targeting native_posix_64, target native_posix instead"
and in the msg below:
"If you were targeting native_posix, target native_posix_64 instead"

Reply:
I see - for some reason I was under the impression that arch/posix was specifically for native_posix. Personally, that's actually all I'm concerned with atm.

Possibly the better solution is to put my changes into `boards/posix/native_posix/` rather than `arch/posix/`, and then just surround the "-m32 / -m64" stuff with a check for `x86_64`. Are there any other arches for `arch/posix` that use "-m32 / -m64"?

Reply:
Doing it here is correct. We just need to write a message that does not confuse users too much.

>Are there any other arches for arch/posix

You mean boards? In the tree only native_posix_64 sets CONFIG_64BIT by default. But there is off-tree boards and anyhow users are free to set it differently.

Reply:
@aescolar - I have to remember to get your feedback before respinning the PR ;-) Your feedback is very good, btw.